storage: TestNodeLivenessStatusMap failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/e9ec5a3d827b6b4620b8aeea1851193453ca952c

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1175348&tab=buildLog

```
2      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190313 07:35:01.067051 632586 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190313 07:35:01.074465 633915 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43967: initial connection heartbeat failed: rpc error: code = Canceled desc = context canceled
I190313 07:35:01.080051 632586 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190313 07:35:01.080084 632586 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190313 07:35:01.107212 632586 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190313 07:35:01.107306 632586 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190313 07:35:01.109551 633988 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35001: initial connection heartbeat failed: rpc error: code = Unavailable desc = transport is closing
I190313 07:35:01.112623 632586 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190313 07:35:01.112670 632586 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190313 07:35:01.112685 632586 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190313 07:35:01.119350 632586 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190313 07:35:01.119523 632586 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190313 07:35:01.122770 634029 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35001: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:35001: connect: connection refused"
I190313 07:35:01.122952 634030 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43967: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:43967: connect: connection refused"
I190313 07:35:01.125248 632586 storage/node_liveness_test.go:851  [in test] done removing node 4
I190313 07:35:01.125327 632586 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.75s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
I190313 07:35:10.882290 633984 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.10s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 633984 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000f4e7e0, 0x33fe740)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3468020, 0xc00fffd500, 0xc000f4e6f0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00fffd500)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc00fffd500, 0xc0018b5590)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/f95d45653df6be5587fb9887de241f50b6932000

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1186082&tab=buildLog

```
I190319 07:17:01.727392 746226 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190319 07:17:01.727572 746226 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190319 07:17:01.727618 746226 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190319 07:17:01.737430 747619 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39117: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:39117: connect: connection refused"
I190319 07:17:01.753105 746226 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190319 07:17:01.753208 746226 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      ts.poller: poll
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190319 07:17:01.754164 746226 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      ts.poller: poll
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190319 07:17:01.754205 746226 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      ts.poller: poll
1      [async] closedts-rangefeed-subscriber
I190319 07:17:01.758155 747517 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:39117: context canceled
I190319 07:17:01.758241 747516 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41395: context canceled
I190319 07:17:01.758916 747560 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:35133: initial connection heartbeat failed: rpc error: code = Canceled desc = grpc: the client connection is closing
I190319 07:17:01.759286 746226 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      ts.poller: poll
I190319 07:17:01.759361 746226 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      ts.poller: poll
I190319 07:17:01.759408 746226 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
1      [async] closedts-subscription
I190319 07:17:01.759448 746226 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
W190319 07:17:01.768909 747442 ts/db.go:211  [n4,ts-poll] error writing time series data: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
I190319 07:17:01.786067 746226 storage/node_liveness_test.go:851  [in test] done removing node 4
I190319 07:17:01.786103 746226 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.54s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.09s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 747610 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc011fd3b00, 0x3416ca0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3480360, 0xc000ea8000, 0xc011fd2ff0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000ea8000)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc000ea8000, 0xc011fd23c0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/668162cc99e4f3198b663b1abfa51858eeb3ccb8

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1212718&tab=buildLog

```
I190401 05:48:09.307036 765180 sql/event_log.go:135  [n4,client=127.0.0.1:40342,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.range_split.by_load_enabled Value:false User:root}
I190401 05:48:09.307230 764293 storage/node_liveness_test.go:824  [in test] waiting for node statuses
n1: found liveness
n2: found liveness
n3: found liveness
n4: found liveness
I190401 05:48:09.312297 764293 storage/node_liveness_test.go:827  [in test] waiting done
I190401 05:48:09.312318 764293 storage/node_liveness_test.go:834  [in test] shutting down node 2
I190401 05:48:09.312373 764293 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.312722 764293 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.312753 764293 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.315323 764293 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
I190401 05:48:09.315609 764293 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190401 05:48:09.319016 764293 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190401 05:48:09.319037 764293 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190401 05:48:09.324326 764293 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190401 05:48:09.324384 764293 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190401 05:48:09.324400 764293 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190401 05:48:09.327731 764293 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190401 05:48:09.327783 764293 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.328003 764293 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.328029 764293 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190401 05:48:09.328148 764293 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190401 05:48:09.330174 764293 storage/node_liveness_test.go:851  [in test] done removing node 4
I190401 05:48:09.330191 764293 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.49s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.09s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 765471 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc00097cf60, 0x345bba0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34c65a0, 0xc000a97400, 0xc00097cf00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000a97400)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc000a97400, 0xc0009a51d0)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/c6df752eefe4609b8a5bbada0955f79a2cfb790e

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1218492&tab=buildLog

```
I190403 06:18:39.015749 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:40.016097 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:40.016460 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:41.016759 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:41.017153 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:42.017464 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:42.017883 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:43.018203 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:43.018584 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:44.018874 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:44.019291 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:45.019622 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:45.020048 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:46.020353 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:46.020816 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:47.021121 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:47.021510 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:48.021898 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:48.022364 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:49.022754 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:49.023089 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:50.023350 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:50.023818 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:51.024197 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:51.024532 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190403 06:18:52.024983 261951 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190403 06:18:52.025328 261951 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 261951 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000be8660, 0x4b14740)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b8cb40, 0xc0014c1a00, 0xc000be8600)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc0014c1a00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc0014c1a00, 0xc008484e10)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/1cbf3680129e47bd310640bf32b665662f30faa9

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1218160&tab=buildLog

```
I190403 06:24:33.131822 499864 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
I190403 06:24:33.131868 499864 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
I190403 06:24:33.131910 499864 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190403 06:24:33.137304 499864 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190403 06:24:33.137367 499864 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190403 06:24:33.162348 501213 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41083: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:41083: connect: connection refused"
I190403 06:24:33.168289 500932 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41083: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:41083: connect: connection refused"
I190403 06:24:33.169601 499864 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190403 06:24:33.169715 499864 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190403 06:24:33.169758 499864 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190403 06:24:33.172250 500956 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:41083: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:41083: connect: connection refused"
I190403 06:24:33.175048 499864 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190403 06:24:33.175246 499864 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190403 06:24:33.183942 499864 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190403 06:24:33.185208 499864 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190403 06:24:33.193260 499864 storage/node_liveness_test.go:851  [in test] done removing node 4
I190403 06:24:33.193327 499864 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.99s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
I190403 06:24:42.824685 501183 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.10s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 501183 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000d24d20, 0x342c520)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3496060, 0xc00b988100, 0xc000d24cc0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00b988100)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc00b988100, 0xc000946d80)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/5267932f6fec0405b31328c1ad43711b0bb013e5

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1220975&tab=buildLog

```
I190404 06:34:23.203428 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:24.203799 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:24.204254 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:25.204696 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:25.205025 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:26.205402 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:26.205841 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:27.206152 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:27.206549 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:28.206837 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:28.207242 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:29.207564 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:29.208059 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:30.208356 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:30.208920 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:31.209292 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:31.209755 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:32.210081 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:32.210479 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:33.211075 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:33.211477 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:34.211872 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:34.212472 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:35.212849 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:35.213232 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190404 06:34:36.213550 258520 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190404 06:34:36.213935 258520 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.12s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 258520 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001714ea0, 0x4b14b80)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b8cf60, 0xc0085dd100, 0xc001714e40)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc0085dd100)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc0085dd100, 0xc007dd2810)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/af621b39a13732f95dde42579b5d6fb55c9e2da7

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1223631&tab=buildLog

```
I190405 06:59:15.628802 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:16.629220 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:16.629650 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:17.630077 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:17.630458 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:18.630896 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:18.631296 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:19.631708 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:19.632042 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:20.632495 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:20.632883 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:21.633265 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:21.633740 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:22.634171 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:22.634587 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:23.635017 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:23.635456 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:24.635807 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:24.636179 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:25.636586 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:25.637057 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:26.637486 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:26.637880 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:27.638360 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:27.638733 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190405 06:59:28.639138 260985 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190405 06:59:28.639540 260985 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.12s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 260985 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001a362a0, 0x4b44c00)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4bbd000, 0xc000761900, 0xc001a36240)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000761900)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000761900, 0xc001518570)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/1a5eabad4511a3371a6b2809d2bfc29e8aff66a6

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1225946&tab=buildLog

```
I190406 07:32:14.388407 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:15.388772 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:15.389211 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:16.389511 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:16.389875 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:17.390192 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:17.390598 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:18.390902 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:18.391326 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:19.391593 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:19.391890 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:20.392169 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:20.392551 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:21.392810 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:21.393260 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:22.393584 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:22.393912 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:23.394288 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:23.394676 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:24.394941 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:24.395265 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:25.395539 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:25.395865 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:26.396147 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:26.396461 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190406 07:32:27.396728 272140 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190406 07:32:27.397067 272140 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 272140 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0012c9830, 0x4b44f60)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4bbd360, 0xc0005c2200, 0xc0012c96b0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc0005c2200)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc0005c2200, 0xc00a044180)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/58c458efeaa3b38c8c982f23a36381aac1b1004b

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1226883&tab=buildLog

```
I190407 05:52:48.634027 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634191 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634342 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634519 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634683 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634830 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.634996 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635122 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635308 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635449 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635573 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635698 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.635926 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.636192 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.636559 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.637016 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.637797 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.639085 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.641414 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.645875 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.654588 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.671619 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.705557 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.772999 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:48.907817 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:49.176567 261940 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190407 05:52:49.633121 261975 storage/node_liveness_test.go:885  [in test] checking expected status for node 4
I190407 05:52:49.633459 261975 storage/node_liveness_test.go:899  [in test] node 4 status ok
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 261940 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc016978480, 0x4b44f60)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4bbd360, 0xc00162d900, 0xc016978420)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00162d900)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc00162d900, 0xc000917ce0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/58c458efeaa3b38c8c982f23a36381aac1b1004b

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1226882&tab=buildLog

```
I190407 06:09:43.611639 647068 storage/node_liveness_test.go:824  [in test] waiting for node statuses
n1: found liveness
n2: found liveness
n3: found liveness
n4: found liveness
I190407 06:09:43.616280 647068 storage/node_liveness_test.go:827  [in test] waiting done
I190407 06:09:43.616302 647068 storage/node_liveness_test.go:834  [in test] shutting down node 2
I190407 06:09:43.616350 647068 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.616377 647068 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.617056 647068 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.617104 647068 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.617181 647068 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190407 06:09:43.618595 647068 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190407 06:09:43.618617 647068 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190407 06:09:43.623063 647068 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190407 06:09:43.623156 647068 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190407 06:09:43.623186 647068 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190407 06:09:43.626887 647068 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190407 06:09:43.626939 647068 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.627164 647068 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.627205 647068 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190407 06:09:43.627354 647068 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
I190407 06:09:43.627437 647068 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190407 06:09:43.631543 647068 storage/node_liveness_test.go:851  [in test] done removing node 4
I190407 06:09:43.631573 647068 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.37s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.09s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 648330 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000cbf290, 0x3489580)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34f4100, 0xc00308a600, 0xc000cbf230)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00308a600)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc00308a600, 0xc015198570)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/c03c307b1a4100a8a1edd2804bdfaf3903097756

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1229620&tab=buildLog

```
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190409 06:41:47.880900 633833 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190409 06:41:47.884952 635050 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35295: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:35295: connect: connection refused"
I190409 06:41:47.885254 633833 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190409 06:41:47.885318 633833 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190409 06:41:47.902246 633833 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190409 06:41:47.902322 633833 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190409 06:41:47.902557 633833 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190409 06:41:47.908829 635276 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:35295: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:35295: connect: connection refused"
I190409 06:41:47.914526 633833 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190409 06:41:47.915118 633833 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190409 06:41:47.916120 633833 util/stop/stopper.go:546  quiescing; tasks left:
3      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190409 06:41:47.916166 633833 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190409 06:41:47.916191 633833 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
I190409 06:41:47.916227 633833 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
I190409 06:41:47.916793 633833 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190409 06:41:47.923417 633833 storage/node_liveness_test.go:851  [in test] done removing node 4
I190409 06:41:47.923440 633833 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (11.07s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.10s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 635334 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc007b44930, 0x345dd80)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34c7b60, 0xc00b2a6b00, 0xc007b44600)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00b2a6b00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc00b2a6b00, 0xc000bdbe90)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/682c2f2f466bbf768545ca4687822206a63983ad

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1232563&tab=buildLog

```
I190410 06:38:57.937243 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:38:58.937544 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:38:58.937931 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:38:59.938211 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:38:59.938597 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:00.938934 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:00.939352 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:01.939686 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:01.940027 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:02.940295 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:02.940747 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:03.941049 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:03.941463 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:04.941794 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:04.942175 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:05.942547 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:05.943019 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:06.943313 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:06.943690 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:07.944019 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:07.944407 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:08.944732 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:08.945090 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:09.945484 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:09.945883 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190410 06:39:10.946340 303932 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190410 06:39:10.946684 303932 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 303932 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001d8bef0, 0x4b4d5e0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4bc59c0, 0xc000d2f900, 0xc001d8bda0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000d2f900)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000d2f900, 0xc000b41740)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/0f5dceb3f1561999f3f8813324eb6099e8030f6e

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1235271&tab=buildLog

```
I190411 06:34:49.374702 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:50.375086 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:50.375504 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:51.375843 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:51.376169 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:52.376506 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:52.376901 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:53.377297 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:53.377617 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:54.377950 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:54.378253 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:55.378587 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:55.378918 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:56.379270 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:56.379646 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:57.380006 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:57.380403 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:58.380826 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:58.381160 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:34:59.381555 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:34:59.381841 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:35:00.382253 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:35:00.382608 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:35:01.383050 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:35:01.383485 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190411 06:35:02.383846 315229 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190411 06:35:02.384190 315229 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.12s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 315229 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc007521c50, 0x4b50ec0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4bc9240, 0xc009224000, 0xc007521bf0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc009224000)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc009224000, 0xc0016eecc0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/58a9bf5992326ca6549cd3933bac175ff28e9bfc

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1238397&tab=buildLog

```
I190412 07:53:56.208873 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:53:57.209195 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:53:57.209619 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:53:58.209945 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:53:58.210384 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:53:59.210739 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:53:59.211236 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:00.211551 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:00.212000 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:01.212310 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:01.212749 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:02.213217 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:02.213618 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:03.213947 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:03.214436 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:04.214765 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:04.215194 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:05.215549 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:05.216008 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:06.216407 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:06.216789 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:07.217040 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:07.217463 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:08.217775 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:08.218230 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190412 07:54:09.218531 307832 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190412 07:54:09.218913 307832 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.12s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 307832 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc004793050, 0x4bb33e0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c2b8e0, 0xc0018cbd00, 0xc004792ff0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc0018cbd00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc0018cbd00, 0xc00109e360)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/c53084cbd4b2be90789652ffb74b3d0c02930f7c

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1239994&tab=buildLog

```
I190413 06:18:40.282742 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:41.283134 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:41.283514 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:42.283877 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:42.284192 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:43.284485 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:43.284883 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:44.285250 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:44.285607 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:45.285935 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:45.286410 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:46.286697 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:46.287173 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:47.287537 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:47.288003 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:48.288367 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:48.288746 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:49.289246 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:49.289612 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:50.289894 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:50.290293 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:51.290577 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:51.290957 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:52.291207 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:52.291571 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190413 06:18:53.291847 313153 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190413 06:18:53.292200 313153 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 313153 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0019b3680, 0x4bb3520)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c2ba20, 0xc00b4f2400, 0xc0019b35f0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00b4f2400)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc00b4f2400, 0xc00bd417a0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/ee451b4f273e93018969bc0edd34e7105625adaf

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1241086&tab=buildLog

```
I190414 06:18:53.196852 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:54.197136 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:54.197508 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:55.197770 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:55.198192 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:56.198459 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:56.198802 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:57.199087 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:57.199480 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:58.199728 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:58.200067 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:18:59.200332 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:18:59.200688 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:00.200947 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:00.201293 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:01.201545 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:01.201885 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:02.202157 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:02.202514 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:03.202810 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:03.203142 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:04.203398 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:04.203780 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:05.204031 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:05.204406 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190414 06:19:06.204654 316221 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190414 06:19:06.205023 316221 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 316221 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0012dbb90, 0x4bb4520)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c2ca40, 0xc001073900, 0xc0012dbb30)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc001073900)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc001073900, 0xc0012da840)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/d3f704f839ccaef7f10c3af48c78a26d390ae1dc

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1241798&tab=buildLog

```
I190415 05:47:09.108691 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:10.108952 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:10.109375 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:11.109697 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:11.110064 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:12.110337 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:12.110813 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:13.111184 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:13.111608 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:14.111958 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:14.112301 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:15.112621 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:15.113080 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:16.113419 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:16.113779 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:17.114103 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:17.114559 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:18.115011 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:18.115381 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:19.115731 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:19.116195 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:20.116526 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:20.116896 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:21.117291 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:21.117836 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190415 05:47:22.118210 297559 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190415 05:47:22.118592 297559 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.12s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 297559 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc00a3362d0, 0x4bb4520)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c2ca40, 0xc000976800, 0xc00a336270)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000976800)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000976800, 0xc00c897c80)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/162528d0429389364f07221e7b818925243d7519

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1243928&tab=buildLog

```
I190416 06:12:32.203381 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:33.203690 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:33.204042 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:34.204361 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:34.204831 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:35.205091 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:35.205440 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:36.205687 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:36.206056 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:37.206336 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:37.206683 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:38.206949 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:38.207300 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:39.207575 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:39.207971 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:40.208263 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:40.208621 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:41.208873 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:41.209286 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:42.209602 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:42.209982 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:43.210282 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:43.210582 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:44.210877 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:44.211194 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190416 06:12:45.211492 324598 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190416 06:12:45.211896 324598 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 324598 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0059b5dd0, 0x4bbae40)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4c33820, 0xc001175b00, 0xc0059b5d70)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc001175b00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc001175b00, 0xc001871c20)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/b1d5ae29d52698485b5398b57178c2ee02b3d49a

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1246250&tab=buildLog

```
I190417 12:24:55.526883 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:24:56.527151 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:24:56.527520 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:24:57.527786 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:24:57.528126 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:24:58.528415 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:24:58.528772 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:24:59.529043 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:24:59.529354 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:00.529635 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:00.530006 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:01.530254 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:01.530584 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:02.530891 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:02.531297 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:03.531644 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:03.531959 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:04.532318 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:04.532728 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:05.533178 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:05.533578 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:06.533872 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:06.534327 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:07.534656 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:07.534978 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190417 12:25:08.535380 310768 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190417 12:25:08.535748 310768 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 310768 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc015b3a090, 0x4c4e6e0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4cc7420, 0xc000ed4400, 0xc015b3a000)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000ed4400)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000ed4400, 0xc000ca3830)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/83de585d331b05a4aa02a65b353bed6bf829b696

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1248042&tab=buildLog

```
I190418 06:18:16.158228 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:17.158537 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:17.158925 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:18.159246 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:18.159614 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:19.159946 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:19.160305 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:20.160728 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:20.161106 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:21.161384 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:21.161777 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:22.162065 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:22.162413 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:23.162763 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:23.163093 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:24.163368 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:24.163774 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:25.164046 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:25.164434 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:26.164832 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:26.165211 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:27.165483 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:27.165873 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:28.166166 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:28.166580 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190418 06:18:29.166872 293032 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190418 06:18:29.167224 293032 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 293032 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0010555c0, 0x4c62080)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4cdb320, 0xc000eee300, 0xc001055560)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000eee300)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000eee300, 0xc001cb0540)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/e1b3d0fdf11783203e76e7a2b3add59e8562a58d

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1249616&tab=buildLog

```
I190419 06:10:14.575642 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:15.575898 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:15.576261 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:16.576556 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:16.576906 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:17.577155 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:17.577508 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:18.577855 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:18.578147 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:19.578384 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:19.578689 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:20.578948 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:20.579331 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:21.579578 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:21.579919 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:22.580283 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:22.580641 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:23.580913 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:23.581330 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:24.581597 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:24.581875 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:25.582141 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:25.582476 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:26.582753 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:26.583068 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190419 06:10:27.583340 307150 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190419 06:10:27.583662 307150 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 307150 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000a8e9f0, 0x4c5ff00)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4cd91a0, 0xc005ed3700, 0xc000a8e7b0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc005ed3700)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc005ed3700, 0xc008ea0330)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/9938cb1a2cca4c0350244f76845f0c61391d44a7

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1249744&tab=buildLog

```
n1: found liveness
n2: found liveness
n3: found liveness
n4: found liveness
I190419 06:44:05.593053 586356 storage/node_liveness_test.go:827  [in test] waiting done
I190419 06:44:05.593072 586356 storage/node_liveness_test.go:834  [in test] shutting down node 2
I190419 06:44:05.593115 586356 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.593258 586356 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.593284 586356 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.596440 586356 util/stop/stopper.go:546  quiescing; tasks left:
2      [async] closedts-subscription
I190419 06:44:05.596481 586356 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190419 06:44:05.599469 586356 storage/node_liveness_test.go:836  [in test] done shutting down node 2
I190419 06:44:05.599496 586356 storage/node_liveness_test.go:839  [in test] decommissioning node 3
I190419 06:44:05.602154 587538 server/server_update.go:67  [n4] no need to upgrade, cluster already at the newest version
I190419 06:44:05.603654 586356 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 3, info: {}
I190419 06:44:05.603700 586356 storage/node_liveness_test.go:843  [in test] done decommissioning node 3
I190419 06:44:05.607952 586356 storage/node_liveness_test.go:846  [in test] decommissioning and shutting down node 4
I190419 06:44:05.618268 586356 sql/event_log.go:135  [in test] Event: "node_decommissioned", target: 4, info: {}
I190419 06:44:05.618310 586356 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.618534 586356 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.619318 586356 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
1      [async] closedts-rangefeed-subscriber
I190419 06:44:05.619407 586356 util/stop/stopper.go:546  quiescing; tasks left:
1      ts.poller: poll
W190419 06:44:05.623846 587515 ts/db.go:211  [n4,ts-poll] error writing time series data: node unavailable; try another peer
I190419 06:44:05.627668 586356 storage/node_liveness_test.go:851  [in test] done removing node 4
I190419 06:44:05.627700 586356 storage/node_liveness_test.go:853  [in test] checking status map
--- FAIL: TestNodeLivenessStatusMap (10.74s)
=== RUN   TestNodeLivenessStatusMap/LIVE
=== PAUSE TestNodeLivenessStatusMap/LIVE
=== CONT  TestNodeLivenessStatusMap/LIVE
I190419 06:44:15.383414 587390 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.09s)
        node_liveness_test.go:874: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 587390 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001208000, 0x3463b60)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34cd960, 0xc0017ae100, 0xc001601f80)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc0017ae100)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0xd4
            testing.tRunner(0xc0017ae100, 0xc013ba1290)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/4b3a1216e3a387aad900e70fde65b97b0fa17a8c

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1252189&tab=buildLog

```
I190420 06:50:05.002203 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:06.002548 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:06.002853 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:07.003117 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:07.003418 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:08.003693 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:08.004056 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:09.004379 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:09.004749 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:10.005077 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:10.005445 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:11.005772 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:11.006189 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:12.006498 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:12.006873 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:13.007130 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:13.007477 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:14.007758 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:14.008145 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:15.008454 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:15.008843 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:16.009193 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:16.009553 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:17.009962 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:17.010332 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190420 06:50:18.010609 298797 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190420 06:50:18.010953 298797 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 298797 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000be1950, 0x4c54dc0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ccb2c0, 0xc00131c700, 0xc000be18f0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00131c700)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc00131c700, 0xc0018d9fb0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/46f8608c4fe2d94b771beb37bcee19136040fd74

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1253828&tab=buildLog

```
I190422 05:44:50.188725 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.188843 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.188940 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.189030 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.189461 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.189582 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.189751 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.189888 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190040 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190128 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190236 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190331 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190514 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.190764 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.191080 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.191519 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.192221 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.193455 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.195751 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.200170 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.208791 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.225766 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.259555 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.326953 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.461555 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:50.730345 305473 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190422 05:44:51.187660 305476 storage/node_liveness_test.go:885  [in test] checking expected status for node 4
I190422 05:44:51.187929 305476 storage/node_liveness_test.go:899  [in test] node 4 status ok
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 305473 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0011b59b0, 0x4c54a40)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ccaf40, 0xc00108a400, 0xc0011b5950)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc00108a400)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc00108a400, 0xc0015ab440)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/ec4728ae986b46d4f57009233b86971198b275ed

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1255947&tab=buildLog

```
I190423 06:48:54.358820 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:48:55.359153 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:48:55.359559 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:48:56.359923 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:48:56.360363 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:48:57.360647 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:48:57.361039 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:48:58.361442 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:48:58.361839 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:48:59.362158 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:48:59.362592 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:00.362907 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:00.363338 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:01.363647 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:01.364050 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:02.364345 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:02.364743 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:03.365088 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:03.365507 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:04.365797 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:04.366129 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:05.366462 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:05.366879 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:06.367185 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:06.367532 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190423 06:49:07.367811 305740 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190423 06:49:07.368234 305740 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 305740 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0008f57a0, 0x4c4e7a0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4cc48c0, 0xc000afcd00, 0xc0008f5740)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000afcd00)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000afcd00, 0xc001238c60)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/d50ebeae81d28087c73cbd1e69db914533d5665d

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestNodeLivenessStatusMap PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1258114&tab=buildLog

```
I190424 06:26:11.855804 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:12.856043 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:12.856459 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:13.856736 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:13.857317 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:14.857612 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:14.858013 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:15.858282 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:15.858673 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:16.858930 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:16.859312 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:17.859596 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:17.859969 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:18.860270 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:18.860681 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:19.860988 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:19.861401 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:20.861767 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:20.862171 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:21.862467 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:21.862820 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:22.863125 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:22.863538 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:23.863810 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:23.864216 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
I190424 06:26:24.864549 299240 storage/node_liveness_test.go:885  [in test] checking expected status for node 1
I190424 06:26:24.864916 299240 storage/node_liveness_test.go:874  SucceedsSoon: unexpected status: got DEAD, expected LIVE
    --- FAIL: TestNodeLivenessStatusMap/LIVE (45.11s)
        <autogenerated>:1: condition failed to evaluate within 45s: unexpected status: got DEAD, expected LIVE
            goroutine 299240 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0012ddb90, 0x4c69260)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4cdfb40, 0xc000b78300, 0xc0012ddb30)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessStatusMap.func1(0xc000b78300)
            	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:874 +0x19e
            testing.tRunner(0xc000b78300, 0xc00ae68b70)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

--- a/pkg/storage/node_liveness_test.go
+++ b/pkg/storage/node_liveness_test.go
@@ -855,11 +855,14 @@ func TestNodeLivenessStatusMap(t *testing.T) {
 	// See what comes up in the status.
 	callerNodeLiveness := firstServer.GetNodeLiveness()
 
-	type expectedStatus struct {
+	type testCase struct {
 		nodeID         roachpb.NodeID
 		expectedStatus storagepb.NodeLivenessStatus
 	}
-	testData := []expectedStatus{
+
+	// Below we're going to check that all statuses converge and stabilize
+	// to a known situation.
+	testData := []testCase{
 		{liveNodeID, storagepb.NodeLivenessStatus_LIVE},
 		{deadNodeID, storagepb.NodeLivenessStatus_DEAD},
 		{decommissioningNodeID, storagepb.NodeLivenessStatus_DECOMMISSIONING},
@@ -867,9 +870,8 @@ func TestNodeLivenessStatusMap(t *testing.T) {
 	}
 
 	for _, test := range testData {
-		t.Run(test.expectedStatus.String(), func(t *testing.T) {
+		t.Run(fmt.Sprintf("n%d->%s", test.nodeID, test.expectedStatus), func(t *testing.T) {
 			nodeID, expectedStatus := test.nodeID, test.expectedStatus
-			t.Parallel()
 
 			testutils.SucceedsSoon(t, func() error {
 				// Ensure that dead nodes are quickly recognized as dead by
@@ -882,21 +884,17 @@ func TestNodeLivenessStatusMap(t *testing.T) {
 				storage.TimeUntilStoreDead.Override(&firstServer.ClusterSettings().SV,
 					storage.TestTimeUntilStoreDead)
 
-				log.Infof(ctx, "checking expected status for node %d", nodeID)
+				log.Infof(ctx, "checking expected status (%s) for node %d", expectedStatus, nodeID)
 				nodeStatuses := callerNodeLiveness.GetLivenessStatusMap()
-				if st, ok := nodeStatuses[nodeID]; !ok {
-					return fmt.Errorf("%s node not in statuses", expectedStatus)
-				} else {
-					if st != expectedStatus {
-						if expectedStatus == storagepb.NodeLivenessStatus_DECOMMISSIONING && st == storagepb.NodeLivenessStatus_DECOMMISSIONED {
-							// Server somehow shut down super-fast. Tolerating the mismatch.
-							return nil
-						}
-						return fmt.Errorf("unexpected status: got %s, expected %s",
-							st, expectedStatus)
-					}
+				st, ok := nodeStatuses[nodeID]
+				if !ok {
+					return errors.Errorf("node %d: not in statuses\n", nodeID)
 				}
-				log.Infof(ctx, "node %d status ok", nodeID)
+				if st != expectedStatus {
+					return errors.Errorf("node %d: unexpected status: got %s, expected %s\n",
+						nodeID, st, expectedStatus,
+					)
+				}
 				return nil
 			})
 		})

